SyntaxInterface/syntaxbridge.cpp

```diff
@@ -0,0 +1,482 @@
+		return;
+	}
+
+	DataSetProvider* provider = DataSetProvider::getProvider(dbInMemory);
+	DataSet* dataset = provider->dataSet();
+
+	dataset->beginBatchedToDB();
+
+	dataset->setColumnCount(syntaxBridgeDataSet->columnCount);
+	dataset->setRowCount(syntaxBridgeDataSet->rowCount);
+
+	for (int colNr = 0; colNr < syntaxBridgeDataSet->columnCount; colNr++)
+	{
+		stringvec values;
+		for (int rowNr = 0; rowNr < syntaxBridgeDataSet->rowCount; rowNr++)
+			values.push_back(syntaxBridgeDataSet->columns[colNr].values[rowNr]);
+		dataset->initColumnWithStrings(colNr, syntaxBridgeDataSet->columns[colNr].name, values, {}, syntaxBridgeDataSet->columns[colNr].name, columnType::unknown, {}, threshold, orderLabelsByValue);
+		dataset->columns()[colNr]->labelsTempCount(); // Needs to do this to set the label temp stuff
+	}
+
+	dataset->endBatchedToDB([](float f) {});
+
+	ColumnEncoder::columnEncoder()->setCurrentNames(dataset->getColumnNames(), true);
+}
+
+const char* STDCALL syntaxBridgeLoadQmlAndParseOptions(const char* moduleName, const char* analysisName, const char* qmlFile, const char* options, const char* version, bool preloadData)
+{
+	if (!init())
+	{
+		Log::log() << "Error during initialization" << std::endl;
+		return "";
+	}
+
+	std::string qmlFileStr		= qmlFile,
+				versionStr		= version,
+				analysisNameStr	= analysisName,
+				moduleNameStr	= moduleName;
+
+
+	AnalysisForm* form = getQmlForm(tq(qmlFileStr));
+
+	if (!form)
+	{
+		Log::log() << "Cannot create QML Form " << qmlFileStr << std::endl;
+		return "";
+	}
+
+	Json::Value parsedOptions;
+	std::string errorMsg;
+
+	if (!form->parseOptions(options, parsedOptions, errorMsg))
+	{
+		Log::log() << "Error when parsing options: " << errorMsg << std::endl;
+		return "";
+	}
+
+	gl_extraEncodings->setCurrentNamesFromOptionsMeta(parsedOptions);
+	gl_dataBridge->updateOptionsAccordingToMeta(parsedOptions);
+	ColumnEncoder::colsPlusTypes analysisColsTypes = ColumnEncoder::encodeColumnNamesinOptions(parsedOptions, preloadData);
+
+	rbridge_setWantedCols(analysisColsTypes);
+
+	static std::string result;
+	result = parsedOptions.toStyledString();
+
+	return result.c_str();
+}
+
+
+const char* STDCALL syntaxBridgeGenerateModuleWrappers(const char* modulePath, bool preloadData)
+{
+	if (!init())
+		return "Error during initialization";
+
+	static std::string result;
+	QString modulePathQ		= tq(modulePath),
+			moduleNameQ;
+
+	QDir moduleDir(modulePathQ);
+
+	if (!moduleDir.exists())
+	{
+		result = fq("Module path not found: " + modulePathQ);
+		return result.c_str();
+	}
+
+	QVector<std::pair<QString, QString> > analyses;
+	QFile qmlDescriptionFile(modulePathQ + "/inst/Description.qml");
+	if (!qmlDescriptionFile.exists())
+	{
+		result = "Description.qml file not found in " + fq(modulePathQ);
+		return result.c_str();
+	}
+
+	QString fileContent;
+	QStringList analysesPart;
+	// TODO: The Description.qml cannot be loaded by the QML Engine, since it does not have access to the JASP.Module
```

Comment:
Why not just add JASP.Module to QMLComponents though?

Because parsing is not the way to go